ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Types.hs

```diff
@@ -821,6 +821,12 @@ data ConnectionManagerTrace peerAddr handlerTrace
   | TrConnectionManagerCounters  !ConnectionManagerCounters
   | TrState                      !(Map peerAddr AbstractState)
   -- ^ traced on SIGUSR1 signal, installed in 'runDataDiffusion'
+  | TrUnexpectedlyMissingConnectionState !(ConnectionId peerAddr)
+  -- ^ This is triggered by a failed lookup on the 'ConnectionManagerState'
+  -- using the 'ConnectionId'\'s peerAddr, during connection handler cleanup
+  -- function. This shouldn't really be possible to achieve and its use is
+  -- mostly for us to be able to be total in our case expression over at
+  -- connection handler cleanup.
```

Comment:
We added the trace because we needed to pattern match on a value that is theoretically impossible to reach (theoretically because it is not enforced by the code). Since the case is a nonsense to reach and we want to be total in the pattern matching the best thing to do is to trace that we somehow reached this unexpectedly.

Reply:
It will be cleaner to define special type for that, than to use a tracer for that.